hazelcast/src/test/java/com/hazelcast/map/PartitionsPredicatePerformanceTest.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map;
+
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import com.hazelcast.aggregation.Aggregators;
+import com.hazelcast.config.Config;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.HazelcastInstanceAware;
+import com.hazelcast.internal.util.concurrent.BusySpinIdleStrategy;
+import com.hazelcast.internal.util.concurrent.IdleStrategy;
+import com.hazelcast.partition.PartitionService;
+import com.hazelcast.query.Predicate;
+import com.hazelcast.query.Predicates;
+import com.hazelcast.spi.properties.ClusterProperty;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import com.hazelcast.test.annotation.NightlyTest;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({NightlyTest.class})
+public class PartitionsPredicatePerformanceTest extends HazelcastTestSupport {
```

Comment:
I am not sure how running this test in nightly will be useful. I don't think anyone will take a look at the logs produced by this test and compare it to some baseline etc.

Reply:
Ok, are you suggesting I remove it or change the category?

Reply:
I believe the benchmark you have written is definitely useful, so I wouldn't recommend removing it altogether. I would just remove the test related annotations
```
@RunWith(HazelcastParallelClassRunner.class)
@Category({NightlyTest.class})
```
so that the test runner wouldn't pick it, Maybe even add a main method to it, instead of this test structure. I see that it is done for a few other files like `MapWordCountAggregationBenchmark` or `ClientQueuePerformanceBenchmark`

Reply:
Minor: Maybe we can do this as well